src/TurbulenceClosures/diffusivity_fields.jl

```diff
@@ -0,0 +1,70 @@
+# Fallback constructor for diffusivity types without precomputed diffusivities.
+DiffusivityFields(arch::AbstractArchitecture, grid::AbstractGrid, args...) = nothing
+
+# For a tuple of closures
+DiffusivityFields(
+    arch::AbstractArchitecture, grid::AbstractGrid, tracers,
+    bcs::NamedTuple, closure_tuple::Tuple
+) = Tuple(DiffusivityFields(arch, grid, tracers, bcs, closure) for closure in closure_tuple)
+
+#####
+##### For closures that only require an eddy viscosity νₑ field.
+#####
+
+const NuClosures = Union{AbstractSmagorinsky, AbstractLeith}
+
+DiffusivityFields(
+    arch::AbstractArchitecture, grid::AbstractGrid, tracers, ::NuClosures;
+    νₑ = CellField(arch, grid, DiffusivityBoundaryConditions(grid), zeros(arch, grid))
+) = (νₑ=νₑ,)
+
+function DiffusivityFields(arch::AbstractArchitecture, grid::AbstractGrid, tracers,
+                           bcs::NamedTuple, ::NuClosures)
+    νₑ_bcs = :νₑ ∈ keys(bcs) ? bcs[:νₑ] : DiffusivityBoundaryConditions(grid)
+    νₑ = CellField(arch, grid, νₑ_bcs, zeros(arch, grid))
+    return (νₑ=νₑ,)
+end
+
+#####
+##### For closures that also require tracer diffusivity fields κₑ on each tracer.
+#####
+
+const NuKappaClosures = Union{VAMD, RAMD}
+
+function KappaFields(arch, grid, tracer_names; kwargs...)
+    κ_fields =
+        Tuple(c ∈ keys(kwargs) ?
+              kwargs[c] :
+              CellField(arch, grid, DiffusivityBoundaryConditions(grid), zeros(arch, grid))
+              for c in tracer_names)
+    return NamedTuple{tracer_names}(κ_fields)
+end
+
+function KappaFields(arch, grid, tracer_names, bcs::NamedTuple)
```

Comment:
could also use `TracerDiffusivityFields` and `ViscosityFields` --- it may be more explicit and clear.

Reply:
Done.